src/main.ts

```diff
@@ -99,6 +113,111 @@ async function run() {
   }
 }
 
+// Check if the source request issue is a duplicate.
+async function checkForDuplicates() {
+  const duplicateCheckEnabled = core.getInput('duplicate-check-enabled');
+
+  if (duplicateCheckEnabled !== 'true') {
+    core.info('The duplicate check is disabled');
+    return;
+  }
+
+  const payload = github.context.payload as IssuesOpenedEvent;
+
+  if (payload.action !== 'opened' || !payload.issue) {
+    core.info('Irrelevant action trigger');
+    return;
+  }
+
+  if (!payload.sender) {
+    throw new Error('Internal error, no sender provided by GitHub');
+  }
+
+  const issue = payload.issue as Issue;
+  const labelToCheck = core.getInput('duplicate-check-label', {required: true});
+  const hasTheLabel = issue.labels?.find(label => label.name === labelToCheck);
+
+  if (!hasTheLabel) {
+    core.info('The issue does not have the label defined');
+    return;
+  }
+
+  const issueUrls = urlsFromIssueBody(issue.body)
+
+  if (issueUrls.length === 0) {
+    core.info('No URLs found in the issue body');
+    return;
+  }
+
+  const client = github.getOctokit(
+    core.getInput('repo-token', {required: true})
+  );
+
+  const { repo } = github.context;
+
+  const allOpenIssues = await client.paginate(client.rest.issues.listForRepo, {
```

Comment:
Does this get all the pages internally?

Reply:
It gets the issues of the next pages when needed, i.e. when it consumes all the `per_page` items in the current page.